src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -109,39 +106,18 @@ public unsafe static SocketError Accept(SafeCloseSocket socketHandle, byte[] soc
 
         private void InnerReleaseHandle()
         {
-            // No-op for Unix.
+            if (_asyncContext != null)
```

Comment:
Again, this was just moved from elsewhere.  I'm not sure a volatile read is needed here, both due to the data-dependency here and due to synchronization elsewhere.